### PR TITLE
[compoundmedia] Add new extractor

### PR DESCRIPTION
**Pull request by kevgrig**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

New extractor for [https://www.compoundmedia.com/](https://www.compoundmedia.com/). No tests because all videos require authentication.

### CODE REVIEW

File: youtube_dl/extractor/compoundmedia.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    # No tests because all videos require authentication
+    # _TEST = {
+    #     'url': 'https://www.compoundmedia.com/shows/tacs/1851',
+    #     'md5': '756a17ea88a671d9faafc6aa406abdb0',
+    #     'info_dict': {
+    #         'id': '1851',
+    #         'title': 'TACS 001 - The First Episode',
+    #     }
+    # }
```
Comment (dstftw) [inline youtube_dl/extractor/compoundmedia.py]: Remove all garbage.

File: youtube_dl/extractor/compoundmedia.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        url = metadata['m3u8']
+        url = url.replace(
+            "https://livestreamapis.com/v3", "https://livestream.com")
+        url = url.replace(".m3u8", "/player")
+
+        title = metadata['title']
+
+        self.to_screen("Title: " + title)
+        self.to_screen("Requesting " + url)
```
Comment (dstftw) [inline youtube_dl/extractor/compoundmedia.py]: Remove.

File: youtube_dl/extractor/compoundmedia.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        livestreamie = LivestreamIE()
```
Comment (dstftw) [inline youtube_dl/extractor/compoundmedia.py]: No, delegation only via `url_result`.

File: youtube_dl/extractor/compoundmedia.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                value, domain = cfduid.groups()
+                self._set_cookie(domain, '__cfduid', value)
+                break
+
+        metadata = self._download_json(
+            self._METADATA_URL_PREFIX + video_id, None, 'Requesting Metadata')
+
+        url = metadata['m3u8']
+        url = url.replace(
+            "https://livestreamapis.com/v3", "https://livestream.com")
```
Comment (dstftw) [inline youtube_dl/extractor/compoundmedia.py]: Single quotes.

File: youtube_dl/extractor/compoundmedia.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            if header.lower() != 'set-cookie':
+                continue
+            if sys.version_info[0] >= 3:
+                cookies = cookies.encode('iso-8859-1')
+            cookies = cookies.decode('utf-8')
+            cfduid = re.search(
+                r'__cfduid=(.+?);.*?\bdomain=(.+?)(?:[,;]|$)', cookies)
+            if cfduid:
+                value, domain = cfduid.groups()
+                self._set_cookie(domain, '__cfduid', value)
```
Comment (dstftw) [inline youtube_dl/extractor/compoundmedia.py]: What's the point of this? `set-cookie` headers are handled internally.

Comment (kevgrig) [reply youtube_dl/extractor/compoundmedia.py]: @dstftw I was just copying from another. So all Set-Cookie's are persisted for future requests?

Comment (dstftw) [reply youtube_dl/extractor/compoundmedia.py]: Have you even bothered to read the rationale comment from where you copy pasted it?

Comment (kevgrig) [reply youtube_dl/extractor/compoundmedia.py]: @dstftw No, sorry. I've removed the code and it works.

**Comment by kevgrig** [comment]
@dstftw Hi, I believe all the requested changes are resolved and the Travis build passed. Can you please re-review?

**Comment by kevgrig** [comment]
@dstftw I wasn't aware of that requirement. I don't like the security and privacy implications of that. Thanks for your consideration.

For any others interested, I simply patched my distribution-installed package:

```
$ echo "from .compoundmedia import CompoundMediaIE" | sudo tee -a /usr/lib/python3.7/site-packages/youtube_dl/extractor/extractors.py
$ sudo curl -o /usr/lib/python3.7/site-packages/youtube_dl/extractor/compoundmedia.py https://raw.githubusercontent.com/rg3/youtube-dl/1f27d68da21a251b0c57e5e6dbba04a2f68414d2/youtube_dl/extractor/compoundmedia.py
$ cat >> ~/.netrc
machine compoundmedia login $EMAIL password $PASSWORD
Ctrl^D
$ youtube-dl -n https://www.compoundmedia.com/shows/...
[...]
[CompoundMedia] Logging in
[CompoundMedia] Requesting Metadata
[livestream] [...]: Downloading JSON metadata
[livestream] [...]: Downloading SMIL file
[livestream] [...]: Downloading m3u8 information
[livestream] [...]: Downloading f4m manifest
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: [...]
[download] Destination: [...].mp4
[download] 100% of [...]MiB in 01:35
[ffmpeg] Fixing malformed AAC bitstream in "[...].mp4"
```

**Pull request closed without merging**